--- cmake/ecbuild_get_test_data.cmake	2017-12-20 09:46:04.708374730 +0100
+++ cmake/ecbuild_get_test_data-patched.cmake	2017-12-20 10:17:52.683511623 +0100
@@ -12,60 +12,7 @@
 
 function( _download_test_data _p_NAME _p_DIRNAME )
 
-  # TODO: make that 'at ecmwf'
-  #if(1)
-  #unset(ENV{no_proxy})
-  #unset(ENV{NO_PROXY})
-  #set(ENV{http_proxy} "http://proxy.ecmwf.int:3333")
-  #endif()
-
-  # Do not retry downloads by default (ECBUILD-307)
-  if( NOT DEFINED ECBUILD_DOWNLOAD_RETRIES )
-    set( ECBUILD_DOWNLOAD_RETRIES 0 )
-  endif()
-  # Use default timeout of 30s if not specified (ECBUILD-307)
-  if( NOT DEFINED ECBUILD_DOWNLOAD_TIMEOUT )
-    set( ECBUILD_DOWNLOAD_TIMEOUT 30 )
-  endif()
-
-  find_program( CURL_PROGRAM curl )
-  mark_as_advanced(CURL_PROGRAM)
-
-  if( CURL_PROGRAM )
-
-    add_custom_command( OUTPUT ${_p_NAME}
-      COMMENT "(curl) downloading http://download.ecmwf.org/test-data/${_p_DIRNAME}/${_p_NAME}"
-      COMMAND ${CURL_PROGRAM} --silent --show-error --fail --output ${_p_NAME}
-              --retry ${ECBUILD_DOWNLOAD_RETRIES}
-              --connect-timeout ${ECBUILD_DOWNLOAD_TIMEOUT}
-              http://download.ecmwf.org/test-data/${_p_DIRNAME}/${_p_NAME} )
-
-  else()
-
-    find_program( WGET_PROGRAM wget )
-
-    if( WGET_PROGRAM )
-
-      # wget takes the total number of tries, curl the number or retries
-      math( EXPR ECBUILD_DOWNLOAD_RETRIES "${ECBUILD_DOWNLOAD_RETRIES} + 1" )
-
-      add_custom_command( OUTPUT ${_p_NAME}
-        COMMENT "(wget) downloading http://download.ecmwf.org/test-data/${_p_DIRNAME}/${_p_NAME}"
-        COMMAND ${WGET_PROGRAM} -nv -O ${_p_NAME}
-                -t ${ECBUILD_DOWNLOAD_RETRIES} -T ${ECBUILD_DOWNLOAD_TIMEOUT}
-                http://download.ecmwf.org/test-data/${_p_DIRNAME}/${_p_NAME} )
-
-    else()
-
-      if( WARNING_CANNOT_DOWNLOAD_TEST_DATA )
-        ecbuild_warn( "Couldn't find curl neither wget -- cannot download test data from server.\nPlease obtain the test data by other means and pleace it in the build directory." )
-        set( WARNING_CANNOT_DOWNLOAD_TEST_DATA 1 CACHE INTERNAL "Couldn't find curl neither wget -- cannot download test data from server" )
-        mark_as_advanced( WARNING_CANNOT_DOWNLOAD_TEST_DATA )
-      endif()
-
-    endif()
-
-  endif()
+    # do nothing :)
 
 endfunction()
 
