--- eccodes-2.12.0-Source.orig/cmake/pkg-config.pc.in	2019-02-15 10:44:50.000000000 +0100
+++ eccodes-2.12.0-Source/cmake/pkg-config.pc.in	2019-02-17 17:16:25.490537820 +0100
@@ -14,9 +14,7 @@ CC=@CMAKE_C_COMPILER@
 CXX=@CMAKE_CXX_COMPILER@
 FC=@CMAKE_Fortran_COMPILER@
 
-rpath=@RPATH_FLAG@${libdir}
-
-libs=-L${libdir} ${rpath} @PKGCONFIG_LIBS@
+libs=-L${libdir} @PKGCONFIG_LIBS@
 
 libs_private=@PKGCONFIG_LIBS_PRIVATE@
 
