--- eccodes-2.17.0-Source.unchanged/src/grib_api_prototypes.h	2020-03-09 11:17:48.000000000 +0100
+++ eccodes-2.17.0-Source/src/grib_api_prototypes.h	2020-03-14 11:59:43.493457319 +0100
@@ -873,8 +873,8 @@ double grib_ieeefloat_error(double x);
 double grib_long_to_ieee(unsigned long x);
 int grib_nearest_smaller_ieee_float(double a, double* x);
 unsigned long grib_ieee_to_long(double x);
-unsigned long grib_ieee64_to_long(double x);
-double grib_long_to_ieee64(unsigned long x);
+uint64_t grib_ieee64_to_long(double x);
+double grib_long_to_ieee64(uint64_t x);
 int grib_ieee_decode_array(grib_context* c, unsigned char* buf, size_t nvals, int bytes, double* val);
 int grib_ieee_decode_array(grib_context* c, unsigned char* buf, size_t nvals, int bytes, double* val);
 int grib_ieee_encode_array(grib_context* c, double* val, size_t nvals, int bytes, unsigned char* buf);
--- eccodes-2.17.0-Source.unchanged/src/grib_ieeefloat.c	2020-03-09 11:17:48.000000000 +0100
+++ eccodes-2.17.0-Source/src/grib_ieeefloat.c	2020-03-14 12:07:45.848470121 +0100
@@ -443,14 +443,14 @@ unsigned long grib_ieee_to_long(double x
  * grib_encode_unsigned_long.
  */
 
-unsigned long grib_ieee64_to_long(double x)
+uint64_t grib_ieee64_to_long(double x)
 {
     unsigned long lval;
     memcpy(&lval, &x, 8);
     return lval;
 }
 
-double grib_long_to_ieee64(unsigned long x)
+double grib_long_to_ieee64(uint64_t x)
 {
     double dval;
     memcpy(&dval, &x, 8);
