--- eccodes-2.25.0-Source.unchanged/src/grib_jasper_encoding.c	2022-03-03 15:40:43.000000000 +0100
+++ eccodes-2.25.0-Source/src/grib_jasper_encoding.c	2022-03-06 16:18:43.875617700 +0100
@@ -30,7 +30,7 @@ int grib_jasper_decode(grib_context* c,
     jas_matrix_t* matrix = NULL;
     jas_image_cmpt_t* p;
     int i, j, k;
-
+    jas_conf_clear(); jas_conf_set_max_mem_usage(jas_get_total_mem_size()); jas_init_library(); jas_init_thread();
     jpeg = jas_stream_memopen((char*)buf, *buflen);
     if (!jpeg) {
         code = GRIB_DECODING_ERROR;
@@ -39,7 +39,7 @@ int grib_jasper_decode(grib_context* c,
 
     grib_context_log(c, GRIB_LOG_DEBUG, "grib_jasper_decode: Jasper version %s", jas_getversion());
 
-    image = jpc_decode(jpeg, NULL);
+    image = jas_image_decode (jpeg, -1, 0);
     if (!image) {
         code = GRIB_DECODING_ERROR;
         goto cleanup;
@@ -77,7 +77,7 @@ cleanup:
         jas_image_destroy(image);
     if (jpeg)
         jas_stream_close(jpeg);
-
+    jas_cleanup_thread(); jas_cleanup_library();
     return code;
 }
 
@@ -94,6 +94,7 @@ int grib_jasper_encode(grib_context* c,
     long no_values         = helper->no_values;
     long bits8;
     int i;
+    int fmt;
 
     size_t buflen          = 0;
     unsigned char* encoded = NULL;
@@ -156,7 +157,7 @@ int grib_jasper_encode(grib_context* c,
         }
     }
 
-    /*jas_init();*/
+    jas_conf_clear(); jas_conf_set_max_mem_usage(jas_get_total_mem_size()); jas_init_library(); jas_init_thread();
 
     opts[0] = 0;
 
@@ -175,7 +176,8 @@ int grib_jasper_encode(grib_context* c,
     cmpt.stream_ = istream;
 
     jpcstream = jas_stream_memopen((char*)helper->jpeg_buffer, helper->buffer_size);
-    jaserr    = jpc_encode(&image, jpcstream, opts);
+    fmt = jas_image_strtofmt("jpc");
+    jaserr    = jas_image_encode(&image, jpcstream, fmt, opts);
 
     if (jaserr != 0) {
         /* increase the number of guard bits */
@@ -189,7 +191,7 @@ int grib_jasper_encode(grib_context* c,
         istream      = jas_stream_memopen((char*)encoded, buflen);
         cmpt.stream_ = istream;
         jpcstream    = jas_stream_memopen((char*)helper->jpeg_buffer, helper->buffer_size);
-        jaserr       = jpc_encode(&image, jpcstream, opts);
+        jaserr       = jas_image_encode(&image, jpcstream, fmt, opts);
     }
 
     if (jaserr != 0) {
@@ -210,7 +212,7 @@ cleanup:
         jas_stream_close(istream);
     if (jpcstream)
         jas_stream_close(jpcstream);
-
+    jas_cleanup_thread(); jas_cleanup_library();
     return code;
 }
 
